packages/workspace/src/utils/buildable-libs-utils.ts

```diff
@@ -46,12 +67,29 @@ export function calculateProjectDependencies(
         depNode.type === ProjectType.lib &&
         isBuildable(context.target.target, depNode)
       ) {
+        const entryFile = resolveEntryFile(context.target.target, depNode);
         const libPackageJson = readJsonFile(
           join(context.workspaceRoot, depNode.data.root, 'package.json')
         );
 
+        // only read this when we have a buildable target since we just need
+        // it for TS Path rewriting
+        if (!tsConfigBaseCache) {
+          const tsConfigPath =
+            target.data.targets[context.target.target].options.tsConfig;
+          tsConfigBaseCache = readTsConfig(tsConfigPath);
+        }
+
+        // find TSConfig mapping based on the sourceRoot
+        const packageImportPath = Object.keys(
+          tsConfigBaseCache?.options?.paths || {}
+        ).find((x) => {
+          const value: string[] = tsConfigBaseCache.options.paths[x];
+          return value.find((t) => t === entryFile);
+        });
+
         return {
-          name: libPackageJson.name, // i.e. @workspace/mylib
+          name: packageImportPath || libPackageJson.name, // i.e. @workspace/mylib
```

Comment:
I think I understand the purpose of the change. You want to point to concrete files so the remapping works. But we use name in other places (e.g., when we rewrite package.json), and the name there has to be the package name. So I don't think you can do an || here.

In general, it's outling the rules for rewriting. I thought we are remapping folders, not files.